npc/re/merchants/pet_trader.txt

```diff
@@ -0,0 +1,177 @@
+//===== rAthena Script =======================================
+//= Pet Trader NPC
+//===== Description: =========================================
+//= [Official Conversion]
+//= Exchanges Cute Pet Tickets for Pet Eggs.
+//===== Additional Comments: =================================
+//= 1.0 First version [Latiosu]
+//============================================================
+
+prontera,210,209,5	script	Pet Trader#1	4_M_JPNOJI,{
+
+	disable_items;
+	mes "[Pet Trader]";
+	mes "Making a living through business ain't easy.";
+	mes "I get real bad motion sickness, yet today I got caught up in a storm on my way here.";
+	next;
+	mes "[Pet Trader]";
+	mes "It was such a horrible experience. Thanks to that, I don't feel so good. Ughh...";
+	next;
+	mes "[Pet Trader]";
+	mes "Anyway, did you come here looking for a Cute Pet?";
+	mes "You've brought a ticket with you, right?";
+	next;
+	switch (select("Examine your tickets:Cancel")) {
+	case 1:
+		.@menu$ = "";
+		.@ticketIndices = 0;
+		.@count = 0;
+		for (.@i = 0; .@i < getarraysize(.tickets); .@i++) {
+			if (countitem(.tickets[.@i]) > 0) {
+				.@menu$ += getitemname(.tickets[.@i]) + ":";
+				.@ticketIndices[.@count++] = .@i;
+			}
+		}
+
+		if (.@count == 0) {
+			mes "[Pet Trader]";
+			mes "I don't think you've got the stuff I want.";
+			close;
+		}
+
+		mes "[Pet Trader]";
+		mes "Which ticket do you want to exchange?";
+		next;
+		.@index = .@ticketIndices[select(.@menu$) - 1];
+
+		mes "[Pet Trader]";
+		mes "Great!";
+		mes "^0000FF"+ getitemname(.eggs[.@index]) +"^000000";
+		mes "Are you sure you want this pet?";
+		next;
+		if (select("Yes:No") == 2) close;
+		if (!checkweight(.eggs[.@index], 1)) {
+			mes "You're carrying too much stuff!";
+			mes "Come back when you've made some space.";
+			close;
+		}
+
+		mes "[Pet Trader]";
+		mes "Here's your pet~";
+		delitem .tickets[.@index], 1;
+		getitem .eggs[.@index], 1;
+		close;
+	case 2:
+		mes "[Pet Trader]";
+		mes "So you want hear about the land across the sea, but I ain't feeling so well today.";
+		close;
+	}
+	end;
+
+OnInit:
+	setarray .tickets,
```

Comment:
Please save the tickets ID in `.@tickets` and eggs ID in `.@eggs` then build the array in "Examine your tickets" menu